linera-storage-service/src/server.rs

```diff
@@ -37,36 +47,43 @@ pub(crate) enum KeyTag {
     Namespace,
 }
 
-pub enum ServiceStoreServer {
+enum ServiceStoreServerInternal {
     Memory(MemoryStore),
     /// The RocksDb key value store
     #[cfg(feature = "rocksdb")]
     RocksDb(RocksDbStore),
 }
 
+struct ServiceStoreServer {
+    store: ServiceStoreServerInternal,
+    index: Arc<RwLock<i64>>,
+    pending_big_puts: Arc<RwLock<BTreeMap<Vec<u8>, Vec<u8>>>>,
+    pending_big_read: Arc<RwLock<BTreeMap<i64, Vec<Vec<u8>>>>>,
+}
```

Comment:
We should probably make these fields LRU caches rather than maps; otherwise if a client cancels a request or is terminated, the value remains in memory forever.

Reply:
I disagree here.
Doing a LRU cache would imply putting a number on the maximal size of the requests.

Yes, there could be some problems with invalidation, but the right solution is the GRPC streams. But since we plan to use it only for tests, that should be fine.

I propose to delay this into some issue to consider in the future: https://github.com/linera-io/linera-protocol/issues/1793